### PR TITLE
[Move CLI] Updated move CLI to work with flattened pkg core Move command

### PR DESCRIPTION
**Pull request by awelc**
This PR updates the move CLI implementation to work with recently updated (https://github.com/move-language/move/pull/240) "flattened" package CLI command implementation in core Move

### CODE REVIEW

File: crates/sui/src/sui_move/build.rs [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
-) -> anyhow::Result<()> {
-    if dump_bytecode_as_base64 {
-        let compiled_modules = sui_framework::build_move_package_to_base64(path, build_config)?;
-        println!("{:?}", compiled_modules);
-    } else {
-        sui_framework::build_and_verify_package(path, build_config)?;
+#[derive(Parser)]
+pub struct Build {
+    #[clap(flatten)]
+    pub new: build::Build,
```
Comment (tnowacki) [inline crates/sui/src/sui_move/build.rs:12]: I feel like this field name is a bit strange

File: crates/sui/src/sui_move/mod.rs [hunk 2]
```diff
@@ -10,46 +11,34 @@ pub mod build;
...
                 ..UnitTestingConfig::default_with_bound(None)
             };
-            let result = unit_test::execute(
-                &package_path,
-                dump_bytecode_as_base64,
-                build_config,
-                unit_test_config,
-                compute_coverage,
-            )?;
+            let result = c.execute(package_path, build_config, unit_test_config)?;
```
Comment (tnowacki) [inline crates/sui/src/sui_move/mod.rs:41]: The building of the unit test config should be done inside execute. And the. You should t need these clones

File: crates/sui/src/sui_move/new.rs [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
+        let name = &self.new.name.to_lowercase();
+        self.new.execute(path,
                         "0.0.1",
-                        "Sui",
-                        "{ git = \"https://github.com/MystenLabs/sui.git\", subdir = \"crates/sui-framework\", rev = \"main\" }",
-                        &name.to_lowercase(),
-                        "0x0")?;
-    Ok(())
+                        [("Sui",
+                          "{ git = \"https://github.com/MystenLabs/sui.git\", subdir = \"crates/sui-framework\", rev = \"main\" }")],
```
Comment (tnowacki) [inline crates/sui/src/sui_move/new.rs:20]: Nit maybe bind to a constant so the formatter doesn't skip this

File: crates/sui/src/sui_move/new.rs [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
-                        name,
+#[derive(Parser)]
+pub struct New {
+    #[clap(flatten)]
+    pub new: new::New,
+}
+
+impl New {
+    pub fn execute(self, path: Option<PathBuf>) -> anyhow::Result<()> {
+        let name = &self.new.name.to_lowercase();
```
Comment (tnowacki) [inline crates/sui/src/sui_move/new.rs:16]: Nit, unpack the command so newly added fields aren't ignored

Comment (awelc) [reply crates/sui/src/sui_move/new.rs:16]: If I unpack than I cannot call `self.new` anymore due to partially moved value and `new::New` on the core Move side is not clone-able - I am not sure how to work around it...

Comment (tnowacki) [reply crates/sui/src/sui_move/new.rs:16]: You can do `let New { name } = &self`

File: crates/sui/src/sui_move/mod.rs (hunk 2 above)
Comment (awelc) [reply crates/sui/src/sui_move/mod.rs:41]: We do add a custom flag to `UnitTestConfig` in crates/sui_framework/src/lib.rs and I am not sure how to do it without constructing the `UnitTestConfig` upfront:
```rust
move_cli::base::test::run_move_unit_tests(
        path,
        build_config,
        UnitTestingConfig {
            report_stacktrace_on_abort: true,
            ..config
        },
        natives::all_natives(MOVE_STDLIB_ADDRESS, SUI_FRAMEWORK_ADDRESS),
        compute_coverage,
        &mut std::io::stdout(),
    )
```

**Review by tnowacki — APPROVED** [review]
Not blocking for nits!

File: crates/sui/src/sui_move/new.rs (hunk 4 above)
Comment (awelc) [reply crates/sui/src/sui_move/new.rs:16]: It still does not work as core Move side's `New::execute` takes `self` by values so we cannot borrow it in the unpack and then use it as the argument...

**Pull request merged**